etl/src/main/java/org/pih/hivmigration/etl/sql/HivExamOisMigrator.groovy

```diff
@@ -0,0 +1,248 @@
+                ('malaria', 'CIEL', '116128'),
+                ('meningitis', 'CIEL', '115835'),
+                ('neurologic_syphilis', 'CIEL','115257'),
+                ('none', 'CIEL', '1007'),
+                ('other', '', ''),
+                ('other_diarrhea', 'CIEL','142412'),
+                ('other_encephalopathy', 'CIEL', '119288'),
+                ('other_meningitis', 'CIEL', '115835'),
+                ('other_neuro', 'PIH', '995'),
+                ('other_pneumonia', 'CIEL', '114100'),
+                ('other_rash', 'CIEL', '512'),
+                ('parasitic_diarrhea', 'PIH','DIARRHEA, PARASITE'),
+                ('pcp_pneumonia', 'CIEL', '882'),
+                ('pid', 'CIEL','902'),
+                ('pneumopathy', '',''),
+                ('prurigo_nodulairis', 'PIH','1370'),
+                ('pruritis', 'PIH', 'PRURITIS'),
+                ('psychosis', 'CIEL','113517'),
+                ('pulmonary_tb', 'CIEL','42'),
+                ('seizures', 'CIEL','113054'),
+                ('shingles', 'CIEL','117543'),
+                ('std', 'PIH', '174'),
+                ('tb_enteritis', 'CIEL','118606'),
+                ('tb_meningitis', 'CIEL','111967'),
+                ('thrush', 'CIEL','5334'),
+                ('toxoplasmosis', 'CIEL','5355'),
+                ('trichomoniasis', 'CIEL','117146'),
+                ('tuberculoma', '', ''),
+                ('tuberculosis', 'CIEL','112141'),
+                ('typhoid', 'CIEL','141'),
+                ('vaginal_urethral_discharge', 'CIEL','123529'),
+                ('vaginal_urethral_gc', '',''),
+                ('vaginal_urethral_other', '',''),
+                ('warts', '',''),
+                ('wasting', 'CIEL','823'),
+                ('weight_loss', 'CIEL','832')                     
+            ''')
+
+        executeMysql("Create staging table for HIV_EXAMS", '''
+            create table hivmigration_hiv_exam_ois (                            
+              obs_id int PRIMARY KEY AUTO_INCREMENT,
+              source_encounter_id int,
+              source_patient_id int,
+              oi VARCHAR(32),
+              comments VARCHAR(128)     
+            );
+        ''')
+
+        setAutoIncrement("hivmigration_hiv_exam_ois", "(select max(obs_id)+1 from obs)")
+
+        loadFromOracleToMySql('''
+            insert into hivmigration_hiv_exam_ois (
+              source_encounter_id,
+              source_patient_id,
+              oi,              
+              comments 
+            )
+            values(?,?,?,?) 
+            ''', '''
+            select 
+                x.ENCOUNTER_ID as source_encounter_id, 
+                e.patient_id as source_patient_id, 
+                lower(trim(x.oi)) as oi,                  
+                x.COMMENTS
+            from HIV_EXAM_OIS x, HIV_ENCOUNTERS e, hiv_demographics_real d  
+            where x.ENCOUNTER_ID = e.ENCOUNTER_ID and e.patient_id = d.patient_id;
+            ''')
```

Comment:
You don't have to, but you can use `(new TableStager("HIV_EXAM_OIS")).migrate()` in place of all this. Or just add it to the TableStagingMigrator.

Reply:
Oh, though you wouldn't then be able to use `obs_id` as the obs group ids. Unless you were to do something like

```
        Object nextId = this.selectMysql(select, new ScalarHandler());
        String sql = "ALTER TABLE " + table + " ADD COLUMN obs_id int primary key auto_increment, auto_increment=" + (nextId != null ? nextId.toString() : "1")
        executeMysql(sql)
```

Reply:
Thanks @brandones ! And, then I would have to add source_patient_id. And the TYPE mapping is not quite correctly. Using TEXT for all VARCHAR makes things very, very slow. The insertion is much faster when I use smaller data types. Thanks!

Reply:
`source_patient_id` is interpolated from `source_encounter_id` by `migrate_tmp_obs`. I didn't realize that about TEXT and VARCHAR! I'll see if I can improve on the mapping. Thanks for the feedback.

Reply:
You could use DATE_LENGTH to determine the length of the defined VARCHAR in the Oracle tables:

select COLUMN_NAME, DATA_TYPE, DATA_LENGTH from ALL_TAB_COLUMNS where TABLE_NAME='HIV_TB_STATUS';

or you could dynamically determine the length of the longest row in a given table:
select max(length(comments)) from HIV_EXAM_OIS;